### PR TITLE
qxmpp/1.4.0

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **qxmpp/1.4.0**

continuation of @COM8's work on https://github.com/conan-io/conan-center-index/pull/6420

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
@COM8 CI is now green. Could you try the recipe and see if it works for you?

**Comment by COM8** [comment]
ACK
Will try it later today. Thanks!

**Review by SSE4 — APPROVED** [review]

File: recipes/qxmpp/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
```
Comment (uilianries) [inline recipes/qxmpp/all/conanfile.py:4]:
```suggestion
required_conan_version = ">=1.33.0"

```

File: recipes/qxmpp/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def validate(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            tools.check_min_cppstd(self, 17)
```
Comment (uilianries) [inline recipes/qxmpp/all/conanfile.py:43]: no minimum compiler version?

File: recipes/qxmpp/all/test_package/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,12 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package)
```
Comment (madebr) [inline recipes/qxmpp/all/test_package/CMakeLists.txt:2]:
```suggestion
project(test_package CXX)
```

File: recipes/qxmpp/all/test_package/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake", "cmake_find_package_multi"
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self.settings):
```
Comment (madebr) [inline recipes/qxmpp/all/test_package/conanfile.py:14]:
```suggestion
        if not tools.cross_building(self):
```

**Comment by COM8** [comment]
Works when building and consuming on Fedora 34 (x64).

**Review by madebr — DISMISSED** [review]

File: recipes/qxmpp/all/test_package/CMakeLists.txt [hunk 5]
```diff
@@ -0,0 +1,12 @@
+cmake_minimum_required(VERSION 3.1)
+project(test_package CXX)
+
+include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+conan_basic_setup(TARGETS)
+
+find_package(QXmpp REQUIRED)
```
Comment (uilianries) [inline recipes/qxmpp/all/test_package/CMakeLists.txt:7]:
```suggestion
find_package(QXmpp REQUIRED CONFIG)
```

The `cmake_find_package_multi` generator creates Cmake config files only, not Findxxx.cmake

**Review by uilianries — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**